src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const el = this[i];
+
+      if (!(el === undefined) && !(el === null)) {
+        result += el;
+      }
+
+      if (!(i === this.length - 1)) {
```

Comment:
Why do you need NOT operator? We have `!==` in JS for such a condition. Or even better - `<`.